src/daemon/distribution.c

```diff
@@ -210,18 +211,16 @@ static void update_tree(distribution_t *dist, size_t node_index, size_t left,
     update_tree(dist, right_child, mid + 1, right, gauge);
 }
 
-void distribution_update(distribution_t *dist, double gauge) {
-  if (dist == NULL)
-    return;
-  if (gauge < 0) {
-    errno = EINVAL;
-    return;
+int distribution_update(distribution_t *dist, double gauge) {
+  if (dist == NULL || gauge < 0) {
+    return errno = EINVAL;
   }
   pthread_mutex_lock(&dist->mutex);
   update_tree(dist, 0, 0, dist->num_buckets - 1, gauge);
   dist->total_sum += gauge;
   dist->total_square_sum += gauge * gauge;
   pthread_mutex_unlock(&dist->mutex);
+  return 0;
```

Comment:
Why did you change the return type to int?

Reply:
I change it in order to return error code if smth went wrong. That case errno can't be overwritten by smth else